### DISCUSSION TITLE
window visualization height Issues

### DISCUSSION THREAD

**Discussion by Coolkeke** [AvaloniaUI/Avalonia#10782] [Q&A]
When Avalonia sets the Window property SizeToContent to WidthAndHeight in a Linux environment, the visual width and height of the current Window do not change when the elements in the Window's Content change。

The current writing method has no problems on Windows environments, except for Linux environments, where the effects are different

The current version is 0.10.18

This is why？

**Comment by timunie** [comment]
1. Linux is sometimes a bit special when it comes to windowing. Some desktopmanagers are quite restricted I think. But anyway, would be interesting to see if it works better using 11.0 preview6

**Comment by timunie** [comment]
From Telegram chat: 

Nikita Tsukanov, [24.03.2023 13:27]
it's impossible to distinguish between user and app triggered Configure events

Nikita Tsukanov, [24.03.2023 13:27]
so size-to-content logic is disabled once the window is shown

Nikita Tsukanov, [24.03.2023 13:27]
it's the same with Qt and other toolkits

**Comment by timunie** [comment]
I'm converting this issue into a Q&A as I think it's not a bug.
